config/routes.rb

```diff
@@ -0,0 +1,73 @@
+Rails.application.routes.draw do
+
+  root 'users#new'
+  resources :profiles, only: [:edit, :update, :show] do
+    get 'timeline'
+  end
+  resources :posts, only: [:index, :create, :destroy] do
```

Comment:
Why do you have index for posts but no action in the controller?

The way I had set up things was a timeline was an index of posts, but you have a timeline inside the profiles controller that seems to do that.